### PR TITLE
mrc-4011 Add step prop to number input

### PR DESCRIPTION
**Pull request by EmmaLRussell**
We need to provide custom step values for MINT as an increment of 1 is not appropriate for all parameter values. This adds optional `step` prop to number inputs. 

Test by viewing `example/index.html` and note that under 'Advanced Options', the 'Decimal parameter' value increments by 0.1 using the spinner arrows.

Some additional housekeeping that was required:
- retire the old travis build and add a github action. Update build status badge on readme (will be updated to master branch on merge). 
- explicitly assert non-nulls in tests
- do not use default coverageProvider - fix for error as described [here](https://github.com/vuejs/vue-jest/issues/450)

### CODE REVIEW

**Review by M-Kusumgar — APPROVED** [review]
Hey emma, nice job, the stepper works as expected. However, while i don't think this is a super important point, it is good to keep in mind: the read only form will only display values with precision of 0.001. So for example if you set the step to 0.0005, it will not accurately display the value. Perhaps that is a note to include in the documentation or something like that. For our purposes however, i don't see that being particularly important!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+name: test
+on: [push]
```
Comment (LekanAnni) [inline .github/workflows/test.yml]:
```
 on: push:
    branches:
      - 'main'
  pull_request:
    branches:
      - 'main'
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 12.x
+      - run: npm install -g codecov
+      - run: npm install
```
Comment (LekanAnni) [inline .github/workflows/test.yml]:
```suggestion
    - run: npm ci
```

**Review by LekanAnni — CHANGES_REQUESTED** [review]
LGTM 

Just some suggestions

**Comment by EmmaLRussell** [comment]
> Hey emma, nice job, the stepper works as expected. However, while i don't think this is a super important point, it is good to keep in mind: the read only form will only display values with precision of 0.001. So for example if you set the step to 0.0005, it will not accurately display the value. Perhaps that is a note to include in the documentation or something like that. For our purposes however, i don't see that being particularly important!

Good point, will add to docs

File: .github/workflows/test.yml (hunk 1 above)
Comment (EmmaLRussell) [reply .github/workflows/test.yml]: Good call, don't need it on every push. But I don't want to limit the PR branches to 'main' as we do sometimes have PRs into non-main branches, so I won't specify branch for `pull-request'

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+name: test
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
```
Comment (LekanAnni) [inline .github/workflows/test.yml]:
```suggestion
  pull_request:
      branches:
         types: [opened, reopened,  edited]
```

**Review by LekanAnni — CHANGES_REQUESTED** [review]
I understand we may need to trigger a job when pull request is opened against other branches. However, you have not specified any target and I do not think you can have an empty `pull_request:` without a target.

File: .github/workflows/test.yml (hunk 3 above)
Comment (LekanAnni) [reply .github/workflows/test.yml]: Maybe you can, just a bit confusing.

**Review by LekanAnni — APPROVED** [review]

File: .github/workflows/test.yml (hunk 3 above)
Comment (EmmaLRussell) [reply .github/workflows/test.yml]: I've put in a ~ which should hopefully make it a bit more obvious that it's an empty 'any' target

**Pull request merged**